### PR TITLE
Feature/person singular

### PR DESCRIPTION
**Pull request by markysoft**
Display 'is' or 'are' depending on number of GPs

### CODE REVIEW

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://profiles-pr-86.dev.c2s.nhschoices.net](http://profiles-pr-86.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://profiles-pr-86.dev.c2s.nhschoices.net](http://profiles-pr-86.dev.c2s.nhschoices.net)

File: test/unit/lib/gpHelper.js [hunk 1]
```diff
@@ -98,4 +98,42 @@ describe('gp counts helper', () => {
...
+    it('should return \'is\' for only one female gp', () => {
+      const gpCounts = {
+        male: 0,
+        female: 1
+      };
+      const personSingular = gpHelper.getPersonSingular(gpCounts);
+      expect(personSingular).to.equal('is');
+    });
+
+    it('should return \'is\' for one female gp and one male GP', () => {
```
Comment (st3v3nhunt) [inline test/unit/lib/gpHelper.js]: Can this test be changed to `should return 'is\ for one female gp and any number of male GPs` to high light the fact that if there is only a single female GP the return will always be `is`. I'd probably have a look within the test that incremented the number of male GPs from 0 to 2 and run the same `expect(...)` as is already here. The test above would then become obsolete.
Another option would be to add an additional test and reword the 3 (this one, the new one and the one above) to be clearer. However, my preference would be the first suggestion.

File: test/unit/lib/gpHelper.js [hunk 2]
```diff
@@ -98,4 +98,42 @@ describe('gp counts helper', () => {
...
+    it('should return \'are\' for more than one male and female gp', () => {
+      const gpCounts = {
+        male: 3,
+        female: 2
+      };
+      const personSingular = gpHelper.getPersonSingular(gpCounts);
+      expect(personSingular).to.equal('are');
+    });
+
+    it('should return \'is\' for only one male gp', () => {
```
Comment (st3v3nhunt) [inline test/unit/lib/gpHelper.js]: Can this be reworded to `should return 'is' for only one male GP and zero female GPs`? This makes it clear that the zero female GPs is important.

File: test/unit/lib/gpHelper.js [hunk 3]
```diff
@@ -98,4 +98,42 @@ describe('gp counts helper', () => {
       expect(message[0]).to.equal('1 GP');
     });
   });
+
+  describe('getPersonSingular', () => {
+    it('should return \'are\' for more than one male and female gp', () => {
```
Comment (st3v3nhunt) [inline test/unit/lib/gpHelper.js]: I'd be tempted to have this test have both GP counts as 2 and reword it to be `should return \'are\' when there are several male GPs and several female GPs`

File: app/middleware/createGpViewModel.js [hunk 4]
```diff
@@ -31,8 +32,8 @@ function createGpViewModel(req, res, next) {
       location: gpData.location,
       facilities: facilitiesMapper(gpData.facilities),
       services: servicesMapper(gpData.services),
-      gpInfo,
-      openingTimes,
+      openingTimes: openingTimesMapper(gpData.openingTimes),
```
Comment (st3v3nhunt) [inline app/middleware/createGpViewModel.js:35]: 👍

File: app/middleware/createGpViewModel.js [hunk 5]
```diff
@@ -31,8 +32,8 @@ function createGpViewModel(req, res, next) {
...
+      gpInfo: getGpInfo(gpData),
```
Comment (st3v3nhunt) [inline app/middleware/createGpViewModel.js:36]: 👍

File: app/views/hub.nunjucks [hunk 6]
```diff
@@ -18,9 +18,9 @@
 
   {% include "includes/patient-survey.nunjucks" %}
 
-  {% include "includes/gps.nunjucks" %}
+  {% include "includes/gp-info.nunjucks" %}
 
-  {% include "includes/services.nunjucks" %}
+  {% include "includes/services.nunjucks" %}
```
Comment (st3v3nhunt) [inline app/views/hub.nunjucks]: Looks like a space has snook onto the end of the line

File: app/views/includes/gp-info.nunjucks [hunk 7]
```diff
@@ -1,14 +1,14 @@
-{% if gp.gpInfo %}
-  <h2>GPs at this surgery</h2>
-  <p>There are
+ {% if gp.gpInfo %}
```
Comment (st3v3nhunt) [inline app/views/includes/gp-info.nunjucks]: Sneaky space at the beginning of the line to remove please

File: app/views/includes/gp-info.nunjucks [hunk 8]
```diff
@@ -1,14 +1,14 @@
...
+  <h2 class='gp-info' >GPs at this surgery</h2>
```
Comment (st3v3nhunt) [inline app/views/includes/gp-info.nunjucks]: Additional space before first `>` can be removed

File: app/views/includes/gp-info.nunjucks [hunk 9]
```diff
@@ -1,14 +1,14 @@
...
   </p>
-    <ul class="link-list">
+    <ul class="link-list gp-info">
       {% for doctor in gp.gpInfo.doctors %}
           <li>{{doctor}}</li>
       {% endfor %}
     </ul>
   <hr class="hr--fullwidth"/>
-{% endif %}
+{% endif %}
```
Comment (st3v3nhunt) [inline app/views/includes/gp-info.nunjucks]: New line please

File: app/lib/gpHelper.js [hunk 10]
```diff
@@ -1,3 +1,13 @@
+function getPersonSingular(gpCounts) {
+  if (gpCounts.female === 1) {
```
Comment (st3v3nhunt) [inline app/lib/gpHelper.js]: The function body could be rewritten as:
```
return gpCounts.female === 1 || (gpCounts.female === 0 && gpCounts.male === 1)
  ? 'is' : 'are';
```
Which I find easier/quicker to read

**Review by st3v3nhunt — DISMISSED** [review]

File: test/unit/lib/gpHelper.js (hunk 3 above)
Comment (markysoft) [reply test/unit/lib/gpHelper.js]: I've changed the GP counts to both be 2, but left the text as 'more than one' rather than several so it is consistently worded with the pluralise tests.

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://profiles-pr-86.dev.c2s.nhschoices.net](http://profiles-pr-86.dev.c2s.nhschoices.net)

**Comment by beseven** [comment]
LOTM!

**Review by beseven — APPROVED** [review]
